sphinx/tutorial/rose/furthertopics/trigger.rst

```diff
@@ -0,0 +1,241 @@
+and ``namelist:pizza_order=no_mushrooms`` is ``.false.``.
+
+Save the metadata file and reload the metadata in the config editor, and
+test it for yourself.
+
+
+Cascading Triggering
+--------------------
+
+Triggering is not just based on values - if a setting is missing or
+``trigger-ignored``, any settings that it triggers will be
+``trigger-ignored`` by default.
+
+This is another way of saying if something is irrelevant, all the settings
+that depend on it should also be irrelevant. This means that triggers can
+act in a cascade - A triggers B triggers C.
+
+We can see this by replacing the ``env=BUDGET`` trigger with:
+
+.. code-block:: cylc
+
+   trigger=namelist:pizza_order=truffle: this > 25;
+           namelist:side_order: this >= 10;
+           namelist:pizza_order=pizza_type: this >= 5;
+
+When ``env=BUDGET`` is less than 5, ``namelist:pizza_order=pizza_type``
+will be ``trigger-ignored``. This means that all of its triggered
+settings like ``namelist:pizza_order=extra_chicken`` are irrelevant and
+will also be ``trigger-ignored``.
+
+We need to add no_mushrooms to the ``[namelist:pizza_order=pizza_type]``
+section so that it is ``trigger-ignored`` when no pizza can be ordered -
+replace the ``[namelist:pizza_order=pizza_type]`` trigger with:
+
+.. code-block:: cylc
+
+   trigger=namelist:pizza_order=extra_chicken: 'BBQ Chicken';
+           namelist:pizza_order=pepperoni_multiple: 'Pepperoni', 'BBQ Chicken';
+           namelist:pizza_order=no_mushrooms;
+
+Save, reload, and try changing ``env=BUDGET`` below 5 to see what it does
+to the options in ``namelist:pizza_order``.
+
+
+Triggering Based On State
+-------------------------
+
+There's also another way to express a trigger - you don't have to express
+a value or range of values in a trigger expression.
+
+Quite often you only want a setting to be ``trigger-ignored`` or enabled
+purely based on the availability of another setting - whether it is present
+and whether it is ``trigger-ignored``. You might not care what particular
+value it has.
+
+This can be expressed by adding a trigger but omitting the value part of
+the syntax. Let's add an option that we can use.
+
+Add a new variable in the metadata by adding these lines to the metadata
+file:
+
+.. code-block:: cylc
+
+   [namelist:pizza_order=dip_type]
+   values='Garlic','Sour Cream','Salsa','Brown Sauce','Mustard'
+
+We should add a trigger expression as well - replace the
+``[namelist:pizza_order=pizza_type]`` trigger with:
+
+.. code-block:: cylc
+
+   trigger=namelist:pizza_order=extra_chicken: 'BBQ Chicken';
+           namelist:pizza_order=pepperoni_multiple: 'Pepperoni', 'BBQ Chicken';
+           namelist:pizza_order=no_mushrooms;
+           namelist:pizza_order=dip_type;
+
+This means that ``namelist:pizza_order=dip_type`` is dependent on
+``namelist:pizza_order=pizza_type``, and will only be ignored when that
+is ignored - but the value of ``pizza_type`` doesn't matter to it.
```

Comment:
Not the clearest (dependency being a specific terminology usually relating to variables and their variation to scientists/mathematicians, the majority of our users I imagine). Consider rephrasing to make it plain that by 'dependent' you mean with respect to the availability/state but not the value.

Reply:
Added TODO.